components/proxy/src/main/java/com/hotels/styx/infrastructure/FileBackedRegistry.java

```diff
@@ -35,95 +38,96 @@
  */
 public class FileBackedRegistry<T extends Identifiable> extends AbstractRegistry<T> {
     private static final Logger LOG = getLogger(FileBackedRegistry.class);
-    private final Parser<T> parser;
-    private final FileMonitor fileMonitor;
-
-    public FileBackedRegistry(Resource configurationFile, Parser<T> parser) {
-        this(parser, configurationFile.absolutePath(), path -> {
-            try {
-                return toByteArray(configurationFile.inputStream());
-            } catch (IOException e) {
-                throw propagate(e);
-            }
-        });
+    private final Resource configurationFile;
+    private final Reader<T> reader;
+    private HashCode fileHash;
+
+    public FileBackedRegistry(Resource configurationFile, Reader<T> reader) {
+        super("FileBackedRegistry");
+        this.configurationFile = checkNotNull(configurationFile);
```

Comment:
To avoid unnecessary use of third-party code, I would recommend using `java.util.Objects.requireNonNull` over its Guava equivalent where possible.